lenskit-core/src/main/java/org/grouplens/lenskit/vectors/similarity/PearsonCorrelation.java

```diff
@@ -57,6 +58,7 @@ public PearsonCorrelation() {
 
     @Inject
     public PearsonCorrelation(@SimilarityDamping double s) {
+        Preconditions.checkArgument(s >= 0, "negative damping not allowed");
```

Comment:
Is this all the places we use damping, I only counted three, but I feel like there might be more.